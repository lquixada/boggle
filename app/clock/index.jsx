/*
 * Timer
 */
var React = require('react');
var ReactDOM = require('react-dom');
var _ = require('underscore');
var $ = require('jquery');
var style = require('./index.less');
var knob = require('jquery-knob');

var ClockView = React.createClass({
  getInitialState: function () {
    var timer = new Timer();
    return {
      secs: timer.remaining,
      timer: timer
    };
  },

  componentWillReceiveProps: function (nextProps) {
    var that = this;

    if (nextProps.started) {
      this.start({
        onStop: function () {
          that.props.onStop();
          alert('Game over!');
        }
      });
    } else {
      this.stop();
    }
  },

  start: function (options) {
    var that = this;
    var opt = options || {};

    this.state.timer.start({
      onTick: function () {
        that.setState({secs: that.state.timer.remaining});

        if (that.state.secs === 0 && _.isFunction(opt.onStop)) {
          opt.onStop();
        }
      }
    });
  },

  stop: function () {
    this.state.timer.stop();
    this.setState({secs: this.state.timer.remaining});
  },

  getSecs: function () {
    var secs = this.state.secs;

    return (secs<10 ? '0'+secs : secs);
  },

  componentDidMount: function () {
    this.renderDial();
  },

  componentDidUpdate: function () {
    this.updateDial();
  },

  renderDial: function () {
    var timer = ReactDOM.findDOMNode(this.refs.timer);

    $(timer).knob({
      readOnly: true,
      width: 120,
      height: 120,
      min: 0,
      max: 60,
      inputColor: '#fff',
      bgColor: '#6c6',
      fgColor: '#666',
      thickness: '.30',
      rotation: 'anticlockwise'
    });
  },

  updateDial: function () {
    var timer = ReactDOM.findDOMNode(this.refs.timer);

    $(timer).trigger('change');
  },

  render: function () {
    return (
      <div id="clock">
        <input ref="timer" value={this.state.secs} readOnly="true" />
        <span className="micro-counter">Time left: 00:{this.getSecs()}</span>
      </div>
    );
  }
});

function Timer(options) {
  var opt = options || {};

  this.frame = opt.frame || 60;
  this.remaining = this.frame;
}

Timer.prototype = {
  start: function (options) {
    var that = this;

    this.onTick = options.onTick || _.noop;
    this.timer = setInterval(function () {
      that.remaining--;
      that.onTick();

      if (!that.remaining) {
        that.stop();
      }
    }, 1000);
  },

  stop: function () {
    this.remaining = this.frame;
    clearInterval(this.timer);
  }
};

module.exports = ClockView;
