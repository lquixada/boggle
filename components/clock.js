import React, { PropTypes } from 'react';
import { connect } from 'react-redux';

import * as actions from '../actions';
import Timer from '../utils/timer.js';

var timer = new Timer();

class Clock extends React.Component {
  constructor(props) {
    super(props);
    this._onChange = this._onChange.bind(this);
    this.state = {
      secs: timer.remaining
    };
  }

  componentDidMount() {
    timer.on('tick', this._onChange);
  }

  componentWillUnmount() {
    timer.removeListener('tick', this._onChange);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.started) {
      // This if guarantees that React Hot Reloading doesn't affect the timer since
      // it's going to pass here a lot of times
      if (!timer.started) {
        this.start();
      }
    } else {
      this.stop();
    }
  }

  start() {
    timer.start();
  }

  stop() {
    timer.stop();
    this.setState({secs: timer.frame});
  }

  getSecs() {
    var secs = this.state.secs;

    return (secs<10 ? '0'+secs : secs);
  }

  _onChange() {
    if (timer.remaining === 0) {
      this.props.stopGame();
      alert('Game over!');
    }

    this.setState({secs: timer.remaining});
  }

  render() {
    return (
      <div id="clock">
        <svg width="120" height="120">
          <circle r="50" cx="60" cy="60" className="clock clock-gray" />
          <circle r="50" cx="60" cy="60" className={'clock clock-green '+(this.props.started? 'running': '')} />
          <text x="60" y="70" className="counter" textAnchor="middle">
            {this.state.secs}
          </text>
        </svg>
        <span className="micro-counter">Time left: 00:{this.getSecs()}</span>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  started: state.started
});

export default connect(mapStateToProps, actions)(Clock);
