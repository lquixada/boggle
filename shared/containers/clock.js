import React from 'react';
import shallowCompare from 'react-addons-shallow-compare';
import {connect} from 'react-redux';
import Clock from '../components/clock';
import * as actions from '../actions';
import Timer from '../utils/timer';

const timer = new Timer();

class ClockContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      secs: timer.remaining
    };

    this.handleChange = this.handleChange.bind(this);
  }

  shouldComponentUpdate(nextProps, nextState) {
    return shallowCompare(this, nextProps, nextState);
  }

  componentDidMount() {
    timer.on('tick', this.handleChange);
  }

  componentWillUnmount() {
    timer.removeListener('tick', this.handleChange);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.started) {
      // This if guarantees that React Hot Reloading doesn't affect the timer since
      // it's going to pass here a lot of times
      if (!timer.started) {
        this.start();
      }
    } else {
      this.stop();
    }
  }

  // eslint-disable-next-line class-methods-use-this
  start() {
    timer.start();
  }

  stop() {
    timer.stop();
    this.setState({secs: timer.frame});
  }

  handleChange() {
    if (timer.remaining === 0) {
      this.props.stopGame();
      alert('Game over!');
    }

    this.setState({secs: timer.remaining});
  }

  render() {
    return (
      <Clock secs={this.state.secs} started={this.props.started} />
    );
  }
}

const mapStateToProps = ({started}) => ({started});
export default connect(mapStateToProps, actions)(ClockContainer);
