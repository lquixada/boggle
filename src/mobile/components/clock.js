import React from 'react'
import PropTypes from 'prop-types'
import { Text } from 'react-native'
import styled from 'styled-components/native'

export class Clock extends React.Component {
  getSecs () {
    return this.props.secs < 10 ? `0${this.props.secs}` : this.props.secs
  }

  getTimeStyle () {
    return this.props.secs < 10 ? { color: 'red' } : {}
  }

  render () {
    return (
      <Label>
        Time left:{' '}
        <Text style={this.getTimeStyle()}>
          00:{this.getSecs()}
        </Text>
      </Label>
    )
  }
}

const Label = styled.Text`
  color: #fff;
  font-size: 15px;
  font-weight: bold;
`

Clock.propTypes = {
  secs: PropTypes.number.isRequired
}
