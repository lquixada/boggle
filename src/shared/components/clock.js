import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import {flex} from '../helpers';

const Clock = ({secs, total}) => {
  const strokeDasharray = 251;
  const strokeDashoffset = ((secs / total) * strokeDasharray);
  const style = {strokeDashoffset, strokeDasharray};

  return (
    <Wrapper>
      <Svg viewBox="0 0 100 100">
        <Dial r="40" cx="50%" cy="50%" className="clock-gray" />
        <Dial r="40" cx="50%" cy="50%" className="clock-green" style={style} transform="rotate(-90 50 50)" />
        <Counter x="50%" y="60%" textAnchor="middle">
          {secs}
        </Counter>
      </Svg>
      <MicroCounter>Time left: 00:{secs < 10 ? `0${secs}` : secs}</MicroCounter>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  ${flex(1)};
  margin-bottom: 2rem;

  @media (max-width: 540px) {
    ${flex('none')};
    margin-bottom: auto;
  }
`;

const Dial = styled.circle`
  fill: transparent;
  stroke-width: 15;

  &.clock-gray {
    stroke: #666;
  }

  &.clock-green {
    stroke: #6c6;
    stroke-dasharray: 251;
  }
`;

const Svg = styled.svg`
  width: 100%;
  height: auto;

  @media (max-width: 540px) {
    display: none !important;
  }
`;

const Counter = styled.text`
  fill: #fff;
  font-family: Arial, sans-serif;
  font-size: 2.5rem;
  font-weight: bold;
`;

const MicroCounter = styled.span`
  display: none;
  font-size: 1.5rem;
  font-weight: bold;

  @media (max-width: 540px) {
    display: block;
  }
`;

Clock.propTypes = {
  secs: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired
};

export default Clock;
