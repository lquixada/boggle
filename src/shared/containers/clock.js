import React from 'react';
import {connect} from 'react-redux';

import Clock from '../components/clock';
import * as actions from '../actions';
import Timer from '../utils/timer';

const timer = new Timer();

export class ClockContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      secs: timer.remaining,
      frame: timer.frame
    };

    this.handleChange = this.handleChange.bind(this);
  }

  componentDidMount() {
    timer.on('tick', this.handleChange);
  }

  componentWillUnmount() {
    timer.removeListener('tick', this.handleChange);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.started) {
      // This if guarantees that React Hot Reloading doesn't affect the timer since
      // it's going to pass here a lot of times
      if (!timer.started) {
        this.start();
      }
    } else {
      this.stop();
    }
  }

  // eslint-disable-next-line class-methods-use-this
  start() {
    timer.start();
  }

  stop() {
    timer.stop();
    this.setState({secs: timer.frame});
  }

  handleChange() {
    this.setState({secs: timer.remaining}, () => {
      if (this.state.secs === 0) {
        // Give some time for React to render the 0-state clock
        // and then stop the game.
        setTimeout(() => {
          this.props.stopGame();
          this.alert();
        }, 100);
      }
    });
  }

  alert() {
    window.alert('Game over!');
  }

  render() {
    return (
      <Clock secs={this.state.secs} total={this.state.frame} />
    );
  }
}

const mapStateToProps = ({started}) => ({started});
export default connect(mapStateToProps, actions)(ClockContainer);
