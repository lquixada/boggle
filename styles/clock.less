#clock {
  margin-bottom: 2rem;

  .clock {
    fill: transparent;
    stroke-width: 20;

    &.clock-gray {
      stroke: #666;
    }

    &.clock-green {
      stroke: #6c6;
      transform: rotate(-90deg);
      transform-origin: 50% 50%;
      stroke-dasharray: 315;
      stroke-dashoffset: 315;
      transition: stroke-dashoffset 0s;

      &.running {
        stroke-dashoffset: 0;
        transition: stroke-dashoffset 60s linear;
      }
    }
  }

  .counter {
    fill: #fff;
    font-family: Arial, sans-serif;
    font-size: 3rem;
    font-weight: bold;
  }

  .micro-counter {
    display: none;
    font-size: 1.5rem;
    font-weight: bold;
  }
}

@media (max-width: 540px) {
  #clock {
    margin-bottom: auto;
    height: auto;
    float: right;

    svg {
      display: none !important;
    }

    .micro-counter {
      display: block;
    }
  }
}
