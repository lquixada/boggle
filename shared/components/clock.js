import React, {PropTypes} from 'react';
import '../../styles/clock.scss';

const Clock = ({secs, started}) => (
  <div id="clock">
    <svg width="120" height="120">
      <circle r="50" cx="60" cy="60" className="clock clock-gray" />
      <circle r="50" cx="60" cy="60" className={`clock clock-green ${started ? 'running' : ''}`} />
      <text x="60" y="70" className="counter" textAnchor="middle">
        {secs}
      </text>
    </svg>
    <span className="micro-counter">Time left: 00:{secs < 10 ? `0${secs}` : secs}</span>
  </div>
);

Clock.propTypes = {
  started: PropTypes.bool.isRequired,
  secs: PropTypes.number.isRequired
};

export default Clock;
