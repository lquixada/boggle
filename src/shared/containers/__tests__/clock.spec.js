import React from 'react';
import {mount} from 'enzyme';
import {ClockContainer} from '../clock';

jest.useFakeTimers();

describe('<ClockContainer />', () => {
  let component;

  beforeEach(() => {
    const stopGame = jest.fn();

    component = mount(<ClockContainer stopGame={stopGame} />);
    component.instance().alert = jest.fn();
  });

  it('displays 60 by default', () => {
    expect(component.update().find('Clock').prop('secs')).toBe(60);
  });

  it('decrements when the game starts', () => {
    component.setProps({started: true});

    expect(component.update().find('Clock').prop('secs')).toBe(59);

    jest.advanceTimersByTime(1000);
    expect(component.update().find('Clock').prop('secs')).toBe(58);

    jest.advanceTimersByTime(1000);
    expect(component.update().find('Clock').prop('secs')).toBe(57);
  });

  it('stops when time have been elapsed', () => {
    component.setProps({started: true});

    jest.advanceTimersByTime(30000);

    component.setProps({started: false});

    expect(component.update().find('Clock').prop('secs')).toBe(60);
  });

  it('stops when time have been elapsed', () => {
    component.setProps({started: true});

    jest.advanceTimersByTime(60000);
    expect(component.props().stopGame).toHaveBeenCalled();
  });

  it('displays 60 when the game stops', () => {
    component.setProps({started: true});

    jest.advanceTimersByTime(60000);
    expect(component.update().find('Clock').prop('secs')).toBe(0);
  });
});
