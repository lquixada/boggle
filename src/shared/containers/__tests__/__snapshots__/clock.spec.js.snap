// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClockContainer /> decrements when the game starts 1`] = `
<ClockContainer
  started={true}
  stopGame={[MockFunction]}
>
  <Clock
    secs={59}
    total={60}
  >
    <styled.div>
      <div
        className="sc-gqjmRU dPXKxW"
      >
        <styled.svg
          viewBox="0 0 100 100"
        >
          <svg
            className="sc-jTzLTM jFWEQC"
            viewBox="0 0 100 100"
          >
            <styled.circle
              className="clock-gray"
              cx="50%"
              cy="50%"
              r="40"
            >
              <circle
                className="clock-gray sc-VigVT iMLRaJ"
                cx="50%"
                cy="50%"
                r="40"
              />
            </styled.circle>
            <styled.circle
              className="clock-green"
              cx="50%"
              cy="50%"
              r="40"
              style={
                Object {
                  "strokeDasharray": 251,
                  "strokeDashoffset": 246.81666666666666,
                }
              }
              transform="rotate(-90 50 50)"
            >
              <circle
                className="clock-green sc-VigVT iMLRaJ"
                cx="50%"
                cy="50%"
                r="40"
                style={
                  Object {
                    "strokeDasharray": 251,
                    "strokeDashoffset": 246.81666666666666,
                  }
                }
                transform="rotate(-90 50 50)"
              />
            </styled.circle>
            <styled.text
              textAnchor="middle"
              x="50%"
              y="60%"
            >
              <text
                className="sc-fjdhpX kwMnmn"
                textAnchor="middle"
                x="50%"
                y="60%"
              >
                59
              </text>
            </styled.text>
          </svg>
        </styled.svg>
        <styled.span>
          <span
            className="sc-jzJRlG bfEpLC"
          >
            Time left: 00:
            59
          </span>
        </styled.span>
      </div>
    </styled.div>
  </Clock>
</ClockContainer>
`;

exports[`<ClockContainer /> decrements when the game starts 2`] = `
<ClockContainer
  started={true}
  stopGame={[MockFunction]}
>
  <Clock
    secs={58}
    total={60}
  >
    <styled.div>
      <div
        className="sc-gqjmRU dPXKxW"
      >
        <styled.svg
          viewBox="0 0 100 100"
        >
          <svg
            className="sc-jTzLTM jFWEQC"
            viewBox="0 0 100 100"
          >
            <styled.circle
              className="clock-gray"
              cx="50%"
              cy="50%"
              r="40"
            >
              <circle
                className="clock-gray sc-VigVT iMLRaJ"
                cx="50%"
                cy="50%"
                r="40"
              />
            </styled.circle>
            <styled.circle
              className="clock-green"
              cx="50%"
              cy="50%"
              r="40"
              style={
                Object {
                  "strokeDasharray": 251,
                  "strokeDashoffset": 242.63333333333333,
                }
              }
              transform="rotate(-90 50 50)"
            >
              <circle
                className="clock-green sc-VigVT iMLRaJ"
                cx="50%"
                cy="50%"
                r="40"
                style={
                  Object {
                    "strokeDasharray": 251,
                    "strokeDashoffset": 242.63333333333333,
                  }
                }
                transform="rotate(-90 50 50)"
              />
            </styled.circle>
            <styled.text
              textAnchor="middle"
              x="50%"
              y="60%"
            >
              <text
                className="sc-fjdhpX kwMnmn"
                textAnchor="middle"
                x="50%"
                y="60%"
              >
                58
              </text>
            </styled.text>
          </svg>
        </styled.svg>
        <styled.span>
          <span
            className="sc-jzJRlG bfEpLC"
          >
            Time left: 00:
            58
          </span>
        </styled.span>
      </div>
    </styled.div>
  </Clock>
</ClockContainer>
`;

exports[`<ClockContainer /> decrements when the game starts 3`] = `
<ClockContainer
  started={true}
  stopGame={[MockFunction]}
>
  <Clock
    secs={57}
    total={60}
  >
    <styled.div>
      <div
        className="sc-gqjmRU dPXKxW"
      >
        <styled.svg
          viewBox="0 0 100 100"
        >
          <svg
            className="sc-jTzLTM jFWEQC"
            viewBox="0 0 100 100"
          >
            <styled.circle
              className="clock-gray"
              cx="50%"
              cy="50%"
              r="40"
            >
              <circle
                className="clock-gray sc-VigVT iMLRaJ"
                cx="50%"
                cy="50%"
                r="40"
              />
            </styled.circle>
            <styled.circle
              className="clock-green"
              cx="50%"
              cy="50%"
              r="40"
              style={
                Object {
                  "strokeDasharray": 251,
                  "strokeDashoffset": 238.45,
                }
              }
              transform="rotate(-90 50 50)"
            >
              <circle
                className="clock-green sc-VigVT iMLRaJ"
                cx="50%"
                cy="50%"
                r="40"
                style={
                  Object {
                    "strokeDasharray": 251,
                    "strokeDashoffset": 238.45,
                  }
                }
                transform="rotate(-90 50 50)"
              />
            </styled.circle>
            <styled.text
              textAnchor="middle"
              x="50%"
              y="60%"
            >
              <text
                className="sc-fjdhpX kwMnmn"
                textAnchor="middle"
                x="50%"
                y="60%"
              >
                57
              </text>
            </styled.text>
          </svg>
        </styled.svg>
        <styled.span>
          <span
            className="sc-jzJRlG bfEpLC"
          >
            Time left: 00:
            57
          </span>
        </styled.span>
      </div>
    </styled.div>
  </Clock>
</ClockContainer>
`;

exports[`<ClockContainer /> displays 60 by default 1`] = `
<ClockContainer
  stopGame={[MockFunction]}
>
  <Clock
    secs={60}
    total={60}
  >
    <styled.div>
      <div
        className="sc-gqjmRU dPXKxW"
      >
        <styled.svg
          viewBox="0 0 100 100"
        >
          <svg
            className="sc-jTzLTM jFWEQC"
            viewBox="0 0 100 100"
          >
            <styled.circle
              className="clock-gray"
              cx="50%"
              cy="50%"
              r="40"
            >
              <circle
                className="clock-gray sc-VigVT iMLRaJ"
                cx="50%"
                cy="50%"
                r="40"
              />
            </styled.circle>
            <styled.circle
              className="clock-green"
              cx="50%"
              cy="50%"
              r="40"
              style={
                Object {
                  "strokeDasharray": 251,
                  "strokeDashoffset": 251,
                }
              }
              transform="rotate(-90 50 50)"
            >
              <circle
                className="clock-green sc-VigVT iMLRaJ"
                cx="50%"
                cy="50%"
                r="40"
                style={
                  Object {
                    "strokeDasharray": 251,
                    "strokeDashoffset": 251,
                  }
                }
                transform="rotate(-90 50 50)"
              />
            </styled.circle>
            <styled.text
              textAnchor="middle"
              x="50%"
              y="60%"
            >
              <text
                className="sc-fjdhpX kwMnmn"
                textAnchor="middle"
                x="50%"
                y="60%"
              >
                60
              </text>
            </styled.text>
          </svg>
        </styled.svg>
        <styled.span>
          <span
            className="sc-jzJRlG bfEpLC"
          >
            Time left: 00:
            60
          </span>
        </styled.span>
      </div>
    </styled.div>
  </Clock>
</ClockContainer>
`;

exports[`<ClockContainer /> displays 60 when the game stops 1`] = `
<ClockContainer
  started={true}
  stopGame={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
    }
  }
>
  <Clock
    secs={0}
    total={60}
  >
    <styled.div>
      <div
        className="sc-gqjmRU dPXKxW"
      >
        <styled.svg
          viewBox="0 0 100 100"
        >
          <svg
            className="sc-jTzLTM jFWEQC"
            viewBox="0 0 100 100"
          >
            <styled.circle
              className="clock-gray"
              cx="50%"
              cy="50%"
              r="40"
            >
              <circle
                className="clock-gray sc-VigVT iMLRaJ"
                cx="50%"
                cy="50%"
                r="40"
              />
            </styled.circle>
            <styled.circle
              className="clock-green"
              cx="50%"
              cy="50%"
              r="40"
              style={
                Object {
                  "strokeDasharray": 251,
                  "strokeDashoffset": 0,
                }
              }
              transform="rotate(-90 50 50)"
            >
              <circle
                className="clock-green sc-VigVT iMLRaJ"
                cx="50%"
                cy="50%"
                r="40"
                style={
                  Object {
                    "strokeDasharray": 251,
                    "strokeDashoffset": 0,
                  }
                }
                transform="rotate(-90 50 50)"
              />
            </styled.circle>
            <styled.text
              textAnchor="middle"
              x="50%"
              y="60%"
            >
              <text
                className="sc-fjdhpX kwMnmn"
                textAnchor="middle"
                x="50%"
                y="60%"
              >
                0
              </text>
            </styled.text>
          </svg>
        </styled.svg>
        <styled.span>
          <span
            className="sc-jzJRlG bfEpLC"
          >
            Time left: 00:
            00
          </span>
        </styled.span>
      </div>
    </styled.div>
  </Clock>
</ClockContainer>
`;

exports[`<ClockContainer /> stops when time have been elapsed 1`] = `
<ClockContainer
  started={false}
  stopGame={[MockFunction]}
>
  <Clock
    secs={60}
    total={60}
  >
    <styled.div>
      <div
        className="sc-gqjmRU dPXKxW"
      >
        <styled.svg
          viewBox="0 0 100 100"
        >
          <svg
            className="sc-jTzLTM jFWEQC"
            viewBox="0 0 100 100"
          >
            <styled.circle
              className="clock-gray"
              cx="50%"
              cy="50%"
              r="40"
            >
              <circle
                className="clock-gray sc-VigVT iMLRaJ"
                cx="50%"
                cy="50%"
                r="40"
              />
            </styled.circle>
            <styled.circle
              className="clock-green"
              cx="50%"
              cy="50%"
              r="40"
              style={
                Object {
                  "strokeDasharray": 251,
                  "strokeDashoffset": 251,
                }
              }
              transform="rotate(-90 50 50)"
            >
              <circle
                className="clock-green sc-VigVT iMLRaJ"
                cx="50%"
                cy="50%"
                r="40"
                style={
                  Object {
                    "strokeDasharray": 251,
                    "strokeDashoffset": 251,
                  }
                }
                transform="rotate(-90 50 50)"
              />
            </styled.circle>
            <styled.text
              textAnchor="middle"
              x="50%"
              y="60%"
            >
              <text
                className="sc-fjdhpX kwMnmn"
                textAnchor="middle"
                x="50%"
                y="60%"
              >
                60
              </text>
            </styled.text>
          </svg>
        </styled.svg>
        <styled.span>
          <span
            className="sc-jzJRlG bfEpLC"
          >
            Time left: 00:
            60
          </span>
        </styled.span>
      </div>
    </styled.div>
  </Clock>
</ClockContainer>
`;
